src/rust/receiver-mock/src/router/api.rs

```diff
@@ -0,0 +1,48 @@
+pub mod v1 {
+    use actix_http::body::Body;
+    use actix_web::{HttpRequest, Responder};
+    use base64;
+    use serde::Serialize;
+
+    #[derive(Serialize)]
+    struct CollectorRegisterRespone {
+        collector_credential_id: String,
+        collector_credential_key: String,
+        collector_id: String,
+        collector_name: String,
+    }
+
+    pub async fn handler_collector_register(req: HttpRequest) -> impl Responder {
+        let header_value = match req.headers().get("Authorization") {
+            Some(v) => v,
+            None => return actix_http::Response::BadRequest().body(Body::Empty),
+        };
+
+        let val_str = match header_value.to_str() {
+            Ok(v) => v,
+            Err(_) => return actix_http::Response::BadRequest().body(Body::Empty),
+        };
+
+        let val = match val_str.strip_prefix("Basic ") {
+            Some(v) => v,
+            None => return actix_http::Response::BadRequest().body(Body::Empty),
+        };
+
+        // For now the token is only checked if it can be decoded successfully.
+        let _decoded = match base64::decode(val) {
+            Ok(v) => v,
+            Err(_) => return actix_http::Response::BadRequest().body(Body::Empty),
+        };
```

Comment:
Is this how Sumo itself behaves? Some of these look like they should be 401s instead of 400s.

Reply:
Changed status code to more or less resemble what Sumo does. 

Also added some tests.